app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/state/MessageComposerState.kt

```diff
@@ -166,12 +166,22 @@ data class MessageComposerState(
     fun hideAttachmentOptions() = changeAttachmentOptionsVisibility(false)
     private fun changeAttachmentOptionsVisibility(newValue: Boolean) {
         (messageComposeInputState as? MessageComposeInputState.Active)?.let { activeState ->
-            (activeState.type as? MessageComposeInputType.NewMessage)?.let { newMessageType ->
-                messageComposeInputState = activeState.copy(
-                    type = newMessageType.copy(
-                        attachmentOptionsDisplayed = newValue
+            when(val currentType = activeState.type){
```

Comment:
here we could simply update the previously declared `attachmentOptionsDisplayed` value.